tools/external_ic.F90

```diff
@@ -570,8 +570,9 @@ subroutine get_nggps_ic (Atm, fv_domain, dt_atmos )
 
 !--- read in the number of tracers in the NCEP NGGPS ICs
       call read_data ('INPUT/'//trim(fn_gfs_ctl), 'ntrac', ntrac, no_domain=.TRUE.)
-      if (ntrac > ntracers) call mpp_error(FATAL,'==> External_ic::get_nggps_ic: more NGGPS tracers &
-                                 &than defined in field_table '//trim(fn_gfs_ctl)//' for NGGPS IC')
+!mzhang: not in FA
+!      if (ntrac > ntracers) call mpp_error(FATAL,'==> External_ic::get_nggps_ic: more NGGPS tracers &

```

Comment:
Do we know why this test has been added there? Are there any potentially dangerous consequences if ntrac > ntracers?

Reply:
The max number of NGGPS tracers (ntrac) is predefined in gfs_ctrl.nc as 7. I guess that it was the max tracers supported at that time?

Reply:
Yeah, but why did they put the requirement there? Is there anything wrong if the number of tracers in the file is larger than what is used? I don't see what should be wrong, but you never know if there is some code elsewhere that relies on this condition. But maybe other code changes have made this check superfluous. Let's comment it out as you suggested and see what GFDL says when we want to merge it back.

Reply:
I don't know if the modification here will impact other places either.

Reply:
!zhang nwat=4
    if ( Atm%flagstruct%nwat .eq. 4 ) then
       do k=1,npz
            BC_side%q_BC(i,j,k,rainwat) = 0.
             BC_side%q_BC(i,j,k,q_rimef) = 0.
             BC_side%q_BC(i,j,k,snowwat) = 0.
             BC_side%q_BC(i,j,k,graupel) = 0.
                  .......
     q_BC is defined by (:,:,:,1:ntracers). So I don't think snowwat and graupel need to be included here.
     I am not sure if mp_auto_conversion_fa is necessary.